src/state/pools/hooks.ts

```diff
@@ -22,13 +22,16 @@ import {
 } from '.'
 import { State, DeserializedPool, VaultKey } from '../types'
 import { transformPool } from './helpers'
+import { fetchFarmsPublicDataAsync, nonArchivedFarms } from '../farms'
 
 export const useFetchPublicPoolsData = () => {
   const dispatch = useAppDispatch()
   const slowRefresh = useSlowFresh()
 
   useEffect(() => {
     batch(() => {
+      const activeFarms = nonArchivedFarms.filter((farm) => farm.pid !== 0)
+      await dispatch(fetchFarmsPublicDataAsync(activeFarms.map((farm) => farm.pid)))
```

Comment:
I'm not sure if `await` here is respected given that its inside `batch` 🤔 
It might work in a sense that UI updates would be batched but other things executed as stated, but I'm not sure, docs & google don't say much about this.
Also the function doesn't have `async` keyword in front of it, I wonder why linter is ok with it. 🤷‍♂️ 

In any case, maybe its safer to refactor it to await for first dispatch and then call `batch` with the other 2? 🙏

Reply:
Done

Reply:
I'm wondering does the `fetchPoolsPublicDataAsync` and `fetchPoolsStakingLimitsAsync` depends on `fetchFarmsPublicDataAsync`? why we need to await `fetchFarmsPublicDataAsync` at the first place? 🤔

Reply:
For Apr calculation, if we don't use await and farms data comes after pools otherwise user has to wait for the next slow refresh cycle to see them which is 1 minute.

Reply:
The issue that i have fixed separately (https://github.com/pancakeswap/pancake-frontend/pull/2863) but it happens with this refactor more often because of less blocking calls so i decided to merge it in here